onebusaway-android/src/main/java/org/onebusaway/android/util/LocationUtils.java

```diff
@@ -407,6 +411,123 @@ public void onConnectionFailed(ConnectionResult connectionResult) {
         return addressesReturn;
     }
 
+    public static List<CustomAddress> processPeliasGeocoding(Context context, ObaRegion region,
+            String... reqs) {
+        return processPeliasGeocoding(context, region, false, reqs);
+    }
+
+    public static List<CustomAddress> processPeliasGeocoding(Context context, ObaRegion region, boolean geocodingForMarker, String... reqs) {
+        ArrayList<CustomAddress> addressesReturn = new ArrayList<CustomAddress>();
+
+        String address = reqs[0];
+
+        if (address == null || address.equalsIgnoreCase("")) {
+            return null;
+        }
+
+        double latitude = 0, longitude = 0;
+        boolean latLngSet = false;
+
+        try {
+            if (reqs.length >= 3) {
+                latitude = Double.parseDouble(reqs[1]);
+                longitude = Double.parseDouble(reqs[2]);
+                latLngSet = true;
+            }
+        } catch (Exception e) {
+            Log.e(TAG, "Geocoding without reference latitude/longitude");
+        }
+
+        if (address.equalsIgnoreCase(context.getString(R.string.tripplanner_current_location))) {
+            if (latLngSet) {
+                CustomAddress addressReturn = new CustomAddress(context.getResources().getConfiguration().locale);
+                addressReturn.setLatitude(latitude);
+                addressReturn.setLongitude(longitude);
+                addressReturn.setAddressLine(addressReturn.getMaxAddressLineIndex() + 1,
+                        context.getString(R.string.tripplanner_current_location));
+
+                addressesReturn.add(addressReturn);
+
+                return addressesReturn;
+            }
+            return null;
+        }
+
+        List<CustomAddress> addresses = new ArrayList<>();
+        try {
+            // FIXME - Replace with Pelias key
+            String apiKey = "YourKeyHere";
```

Comment:
Right now the API key for Pelias is hard-coded, so if you want to test replace `YourKeyHere` with your Pelias API key.

Reply:
Key config has now been moved to gradle.properties - see configuration instructions here - https://github.com/OneBusAway/onebusaway-android/pull/1018/commits/ffca5dcf88e8c25baa7cb67319da0aa33da1aabc.